packages/analytics/src/Providers/AWSPinpointProvider.ts

```diff
@@ -447,7 +447,7 @@ export default class AWSPinpointProvider implements AnalyticsProvider {
 		failureData: EndpointFailureData
 	) {
 		const { err, update_params, endpointObject } = failureData;
-		const { message } = err;
+		const { message = '' } = err;
 		const { ApplicationId, EndpointRequest } = update_params;
 
 		if (!message.startsWith('Exceeded maximum endpoint per user count')) {
```

Comment:
We're going to review this change and confirm the desired logic in the event an `err` comes back without a `message`.

One option is to not default `message` at all, and do a false-y check like so:

```suggestion
		if (!(message && message.startsWith('Exceeded maximum endpoint per user count'))) {
```

Going to mark this for review by the team & get this resolved.

Thanks for sharing your logs: I haven't seen an `err` without a `message` myself 🤞

Reply:
I'm going to update this branch today with a test to validate the changes, then we should be good.

Thanks again for bringing this to our attention @mariotsi.  We've identified this is a `400`, but don't know the full shape of the response.

If you ever _do_ see the shape of the error (e.g. `{ Debug: "Whoopsie!" }`), we can add that into the test.